ledger-service/jwt/src/main/scala/com/digitalasset/jwt/JwtVerifier.scala

```diff
@@ -16,9 +16,14 @@ import scalaz.syntax.traverse._
 
 abstract class JwtVerifierBase {
   def verify(jwt: domain.Jwt): Error \/ domain.DecodedJwt[String]
+
+  def getLeewayOptions: Option[LeewayOptions]
```

Comment:
I am not fond of this mechanism of getting access to the leeway options. It feels like a foreign body in the `JwtVerifier` class hierarchy. I would suggest an attack from another side i.e. the `AuthServiceConfig` class:

```
sealed trait AuthServiceConfig {
  def create(): AuthService
  val leewayOptions: Option[LeewayOptions]
}
```

You should then be able to pass the leewayOptions explicitly to the `ApiServiceOwner` constructor.

You won't then need to modify the `AuthServiceJWT` class, where the changes also feel out of place.

Reply:
In `ApiServiceOwner` the `AuthServiceConfig` is not available. We only can access the `AuthService` class. So then the `AuthService` class must have a new `leewayOptions` field and the constructor of `AuthServiceJWT` must change so as to pass to its parent class (`AuthService`) the `leewayOptions`.

Reply:
`ApiServiceOwner` could be extended to carry `leewayOptions: Option[LeewayOptions]`. This way the parameters are passed top down from the source to the target.

Reply:
Could you have a look at ce0ae0d1388d911a79107517466e0e049f598440?